src/createUniqueTag.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Global counter to scope the custom elements
+ *
+ * @type {number}
+ */
+let counter: number = Math.round(Math.random() * 100000);
+
+/**
+ * Allowed tag name chars
+ *
+ * @type {string}
+ */
+const chars: string = `-|\\.|[0-9]|[a-z]`;
+
+/**
+ * Regular expression to check if a value is a valid tag name
+ *
+ * @type {RegExp}
+ */
+const tagRegExp: RegExp = new RegExp(`[a-z](${chars})*-(${chars})*`);
+
+/**
+ * Checks if the tag name is valid
+ *
+ * @param {string} tag
+ * @returns {boolean}
+ */
+const isValid = (tag: string): boolean => tagRegExp.exec(tag) !== null;
+
+/**
+ * Checks if the tag is already registered
+ *
+ * @param {string} name
+ * @param {CustomElementRegistry} registry
+ * @returns {boolean}
+ */
+const isTagRegistered = (name: string, registry: CustomElementRegistry): boolean => !!registry.get(name);
```

Comment:
I've noticed that you've separated out the registry. I don't know much about this; is it possible to define your own registries, and how often does that happen?

Reply:
Currently there is only 1 global and that's the issue with this registry when you are doing microfrontends.

Specification is still in discussion here: https://github.com/w3c/webcomponents/issues/716